src/engine.rs

```diff
@@ -0,0 +1,106 @@
+use anyhow::{anyhow, Result};
+use std::{
+    path::PathBuf,
+    time::{Duration, Instant},
+};
+use wasmtime::{Engine, Linker, Module, Store};
+
+use wasmtime_wasi::sync::WasiCtxBuilder;
+
+use crate::run_statistics::RunStatistics;
+
+pub fn run(script_path: PathBuf, input_path: PathBuf) -> Result<RunStatistics> {
+    let engine = Engine::default();
+    let module = Module::from_file(&engine, &script_path)
+        .map_err(|e| anyhow!("Couldn't load script {:?}: {}", &script_path, e))?;
+
+    let input: serde_json::Value = serde_json::from_reader(
+        std::fs::File::open(&input_path)
+            .map_err(|e| anyhow!("Couldn't load input {:?}: {}", &input_path, e))?,
+    )
+    .map_err(|e| anyhow!("Couldn't load input {:?}: {}", &input_path, e))?;
+    let input = serde_json::to_vec(&input)?;
+
+    let input_stream = wasi_common::pipe::ReadPipe::new(std::io::Cursor::new(input));
+    let output_stream = wasi_common::pipe::WritePipe::new_in_memory();
+    let error_stream = wasi_common::pipe::WritePipe::new_in_memory();
+
+    let runtime: Duration;
+
+    {
+        let mut linker = Linker::new(&engine);
+        wasmtime_wasi::add_to_linker(&mut linker, |s| s)?;
+        let wasi = WasiCtxBuilder::new()
+            .stdin(Box::new(input_stream))
+            .stdout(Box::new(output_stream.clone()))
+            .stderr(Box::new(error_stream.clone()))
+            .inherit_args()?
+            .build();
+        let mut store = Store::new(&engine, wasi);
+
+        linker.module(&mut store, "", &module)?;
+
+        let start = Instant::now();
+
+        let module_result = linker
+            .get_default(&mut store, "")?
+            .typed::<(), (), _>(&store)?
+            .call(&mut store, ());
+
+        runtime = start.elapsed();
```

Comment:
I'm just wondering if we are measuring the right thing here?  🤔 
Just for context, the purpose of this `script-runner` is so that the devs can measure performance locally, instead of always needing to execute in production.  Therefore, we want to measure as closely as what's done in `runtime-engine`.  Disclaimer, I am not an expert in `runtime-engine`, so these might be dumb questions.  I see that in `runtime-engine`, the timeout is set on the entire module run ([here](https://github.com/Shopify/runtime-engine/blob/2083124ebc748c31dee0d7f2e3d4f948950f8988/http/src/module_executor.rs#L99)).  What you are measuring is more analogous to the `main_duration` done in the finer measurements [here](https://github.com/Shopify/runtime-engine/blob/2083124ebc748c31dee0d7f2e3d4f948950f8988/core/src/module.rs#L124).  I know that this is the only part that's relevant to the dev's code, but should we still measure the entire execution including `start_duration`?


Another question that might be outside the scope of your PR.  @jianghong: do you know if we should emulate the code in the `runtime-engine` a bit closer?  Like in here, I see it's loading the default function, where the `runtime-engine` is looking for a typed function `_start`.  I am not sure if the differences has any performance impact at all.  WDYT?

Reply:
Thanks for the heads up! @jianghong I had to use `store`, because `store.0` was not a thing. Let me know if you know why.

Reply:
Oops yeah, I typo'd there